std/process.d

```diff
@@ -465,80 +490,110 @@ private Pid spawnProcessImpl(in char[][] args,
             close(workDirFD);
         }
 
-        // Redirect streams and close the old file descriptors.
-        // In the case that stderr is redirected to stdout, we need
-        // to backup the file descriptor since stdout may be redirected
-        // as well.
-        if (stderrFD == STDOUT_FILENO)  stderrFD = dup(stderrFD);
-        dup2(stdinFD,  STDIN_FILENO);
-        dup2(stdoutFD, STDOUT_FILENO);
-        dup2(stderrFD, STDERR_FILENO);
-
-        // Ensure that the standard streams aren't closed on execute, and
-        // optionally close all other file descriptors.
-        setCLOEXEC(STDIN_FILENO, false);
-        setCLOEXEC(STDOUT_FILENO, false);
-        setCLOEXEC(STDERR_FILENO, false);
-        if (!(config & Config.inheritFDs))
+        void execProcess()
         {
-            import core.stdc.stdlib : malloc;
-            import core.sys.posix.poll : pollfd, poll, POLLNVAL;
-            import core.sys.posix.sys.resource : rlimit, getrlimit, RLIMIT_NOFILE;
+            // Redirect streams and close the old file descriptors.
+            // In the case that stderr is redirected to stdout, we need
+            // to backup the file descriptor since stdout may be redirected
+            // as well.
+            if (stderrFD == STDOUT_FILENO)  stderrFD = dup(stderrFD);
+            dup2(stdinFD,  STDIN_FILENO);
+            dup2(stdoutFD, STDOUT_FILENO);
+            dup2(stderrFD, STDERR_FILENO);
+
+            // Ensure that the standard streams aren't closed on execute, and
+            // optionally close all other file descriptors.
+            setCLOEXEC(STDIN_FILENO, false);
+            setCLOEXEC(STDOUT_FILENO, false);
+            setCLOEXEC(STDERR_FILENO, false);
+
+            if (!(config & Config.inheritFDs))
+            {
+                import core.stdc.stdlib : malloc;
+                import core.sys.posix.poll : pollfd, poll, POLLNVAL;
+                import core.sys.posix.sys.resource : rlimit, getrlimit, RLIMIT_NOFILE;
+
+                // Get the maximum number of file descriptors that could be open.
+                rlimit r;
+                if (getrlimit(RLIMIT_NOFILE, &r) != 0)
+                {
+                    abortOnError(forkPipeOut, InternalError.getrlimit, .errno);
+                }
+                immutable maxDescriptors = cast(int) r.rlim_cur;
 
-            // Get the maximum number of file descriptors that could be open.
-            rlimit r;
-            if (getrlimit(RLIMIT_NOFILE, &r) != 0)
+                // The above, less stdin, stdout, and stderr
+                immutable maxToClose = maxDescriptors - 3;
+
+                // Call poll() to see which ones are actually open:
+                pollfd* pfds = cast(pollfd*) malloc(pollfd.sizeof * maxToClose);
```

Comment:
Again, you're asking to change things that were like this before me. Well, I can, for sure.